components/org.wso2.carbon.consent.mgt.ui/src/main/resources/web/consent/add-purpose.jsp

```diff
@@ -99,9 +121,30 @@
         function doFinish() {
             document.dataForm.action = "<%=Encode.forJavaScript(addFinishPurposePage)%>";
             if (doValidation() === true) {
+                if(!doValidationForMandatoryEmailP2() && !<%=hasPurposeWithMandatoryEmailInList%> ) {
+                    CARBON.showWarningDialog("<%=resourceBundle.getString("missing.mandatory.email.pii.category.warning.add.purpose")%>",
+                    doSubmit,doSubmit);
```

Comment:
are we not submitting the form when the warning is showing?

Reply:
form is submitted on clicking on ok button or close the warning. if we directly submit, since its redirecting to a different page warning message will not be visible.